src/app/visual/dock/dock-visual.component.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  AfterViewInit,
+  Component
+} from '@angular/core';
+
+import {
+  SkyDockService
+} from '../../public';
+
+import {
+  DockItemVisualContext
+} from './dock-item-context';
+
+import {
+  DockItemVisualComponent
+} from './dock-item-visual.component';
+
+@Component({
+  selector: 'dock-visual',
+  templateUrl: './dock-visual.component.html',
+  styleUrls: ['./dock-visual.component.scss']
+})
+export class DockVisualComponent implements AfterViewInit {
+
+  public stackOrder: number;
+
+  private configs: any[] = [
+    {
+      stackOrder: 0,
+      backgroundColor: 'darkred'
+    },
+    {
+      stackOrder: 100,
+      backgroundColor: 'darkmagenta'
+    },
+    {
+      stackOrder: 10,
+      backgroundColor: 'darkcyan'
+    },
+    {
+      stackOrder: -1000,
+      backgroundColor: 'darkblue'
+    },
+    {
+      stackOrder: 1,
+      backgroundColor: 'darkgreen'
+    }
+  ];
+
+  constructor(
+    private dockService: SkyDockService
+  ) { }
+
+  public ngAfterViewInit(): void {
+    setTimeout(() => {
+      this.configs.forEach((config) => {
+        this.addToDock(config);
+      });
+    });
+  }
+
+  public addItem(): void {
+    let stackOrder = this.stackOrder;
+    if (stackOrder === undefined) {
```

Comment:
Doesn't the `dockService` automatically give you the highest stack order if `stackOrder === undefined` ? Why do this logic in the visual?

Reply:
This was a holdover from a previous experiment; removed.